app/src/main/java/com/aceinteract/android/stepper/presentation/samples/TabStepperWithoutNavigationComponentsActivity.kt

```diff
@@ -0,0 +1,97 @@
+package com.aceinteract.android.stepper.presentation.samples
+
+import android.os.Bundle
+import androidx.appcompat.app.AppCompatActivity
+import androidx.core.view.isVisible
+import com.aceinteract.android.stepper.R
+import com.aceinteract.android.stepper.StepperNavListener
+import kotlinx.android.synthetic.main.tab_stepper_without_navigation_components_activity.*
+import ng.softcom.android.utils.ui.showToast
+
+/**
+ * Activity showing an sample usage of a tab stepper without navigation components.
+ */
+class TabStepperWithoutNavigationComponentsActivity : AppCompatActivity(), StepperNavListener {
+
+    private var animationStartTime = 0L
+    private val menuTitles = arrayListOf<String>()
+
+    /**
+     * Setup stepper and activity.
+     */
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.tab_stepper_without_navigation_components_activity)
+
+        initializeStepper()
+        initializeToolbar()
+        initializeButtons()
+    }
+
+    private fun initializeStepper() {
+        stepper.stepperNavListener = this@TabStepperWithoutNavigationComponentsActivity
+        stepper.setupWithoutNavController()
+    }
+
+    private fun initializeToolbar() {
+        for (i in 0 until stepper.menu.size()) {
+            menuTitles.add(stepper.menu.getItem(i).title.toString())
+        }
+
+        toolbar.title = menuTitles[0]
+        toolbar.setNavigationOnClickListener {
+            onBackPressed()
+        }
+    }
+
+    private fun initializeButtons() {
+        button_next.setOnClickListener {
+            stepper.goToNextStep()
+        }
+
+        button_previous.setOnClickListener {
+            if (stepper.currentStep == 0 && System.currentTimeMillis() - animationStartTime < 300) {
+                return@setOnClickListener
+            }
+            animationStartTime = System.currentTimeMillis()
+            stepper.goToPreviousStep()
+        }
+    }
+
+    override fun onStepChanged(step: Int) {
+        showToast("Step changed to: $step")
+        setupToolbarTitle(step)
+        setupPreviousButtonVisibility(step)
+        setupNextButtonImage(step)
+        setupViewVisibility(step)
+    }
+
+    private fun setupToolbarTitle(step: Int) {
+        toolbar.title = menuTitles[step]
+    }
+
+    private fun setupPreviousButtonVisibility(step: Int) {
+        when (step > 0) {
```

Comment:
You should just use a regular if/else statement here.

E.g:

```kotlin
if (step > 0) button_previous.show()
else button_previous.hide()
```

Or better still, just simplify to a single expression:

```kotlin
button_previous.isVisible = step > 0
```

Reply:
In my opinion when expression looks just cooler and on decompiled to java bytecode it's just regular if/else, I will replace it. isVisible doesn't provide zoomIn/zoomOut animation for fab, but I will replace it too like on other samples.

Reply:
It might be cooler, but it `when` (in this case) uses more code for the same task.

You don't have to use `isVisible`, I just brought that up as an alternative for a simpler expression.